pkg/tempo/config.go

```diff
@@ -77,6 +77,7 @@ type InstanceConfig struct {
 // PushConfig controls the configuration of exporting to Grafana Cloud
 type PushConfig struct {
```

Comment:
Can you implement UnmarshalYAML for this and have it fail if Compression was set to an invalid option? You can also use it to apply a default if it was empty.

Reply:
Sure thing, but want to note that it already fails for invalid options within the exporter.

Reply:
👍 understood, I like keeping as much validation as possible during the unmarshal so config errors don't get buried during startup